vm/runtime.ts

```diff
@@ -1321,47 +1321,47 @@ module J2ME {
    */
   function linkKlassFields(klass: Klass) {
     var classInfo = klass.classInfo;
-    var fields = classInfo.getFields();
-    var classBindings = BindingsMap.get(klass.classInfo.utf8Name);
+    var classBindings = BindingsMap.get(classInfo.utf8Name);
     if (classBindings && classBindings.fields) {
-      for (var i = 0; i < fields.length; i++) {
-        var field = fields[i];
-        // TODO Startup Performance: This iterates over all the fields then looks for symbols
-        // that need to be linked. We should instead scan the symbol list and then look for
-        // matching fields in the class. Doing this will avoid creating the key below that is
-        // only used to lookup symbols.
-        var key = ByteStream.readString(field.utf8Name) + "." + ByteStream.readString(field.utf8Signature);
-        var symbols = field.isStatic ? classBindings.fields.staticSymbols :
-                                       classBindings.fields.instanceSymbols;
-        if (symbols && symbols[key]) {
-          release || assert(!field.isStatic, "Static fields are not supported yet.");
-          var symbolName = symbols[key];
-          var object = field.isStatic ? klass : klass.prototype;
-          release || assert (!object.hasOwnProperty(symbolName), "Should not overwrite existing properties.");
-          var getter = FunctionUtilities.makeForwardingGetter(field.mangledName);
-          var setter;
-          if (release) {
-            setter = FunctionUtilities.makeForwardingSetter(field.mangledName);
-          } else {
-            setter = FunctionUtilities.makeDebugForwardingSetter(field.mangledName, getKindCheck(field.kind));
-          }
-          Object.defineProperty(object, symbolName, {
-            get: getter,
-            set: setter,
-            configurable: true,
-            enumerable: false
-          });
-          delete symbols[key];
+      release || assert(!classBindings.fields.staticSymbols, "Static fields are not supported yet");
+
+      var instanceSymbols = classBindings.fields.instanceSymbols;
+
+      var fieldNames = Object.keys(instanceSymbols);
+
+      for (var i = 0; i < fieldNames.length; i++) {
+        var fieldName = fieldNames[i];
+        var fieldSignature = instanceSymbols[fieldName];
```

Comment:
I did it with Object.keys because I recall something about for ... in being slow, although a simple test (http://jsperf.com/forinobjectkeys) shows that for .. in is faster.